src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -851,30 +854,42 @@ internal SSHConnection[] ParseSSHConnectionHashTable()
                         throw new PSArgumentException(RemotingErrorIdStrings.InvalidSSHConnectionParameter);
                     }
 
-                    string paramValue = item[paramName] as string;
-                    if (string.IsNullOrEmpty(paramValue))
+                    if (paramName.Equals(PortParameter, StringComparison.OrdinalIgnoreCase))
```

Comment:
I didn't like how it was done in code below. We should use the previous style of `if-else if`.

We could even use C# 7.0 patterns.
/cc @daxian-dbw Are we ready to use the patterns?